src/overlays/actors/ovl_En_Bal/z_en_bal.c

```diff
@@ -64,108 +149,1026 @@ static DamageTable D_80A63CE0 = {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bal/func_80A61A44.s")
+                    case TINGLE_MAP_STONE_TOWER:
+                        Message_StartTextbox(play, 0x1D16, &this->picto.actor);
+                        this->prevTextId = 0x1D16;
+                        break;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bal/EnBal_ValidatePictograph.s")
+                    default:
+                        Message_StartTextbox(play, 0x1D11, &this->picto.actor);
+                        this->prevTextId = 0x1D11;
+                        break;
+                }
+                break;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bal/func_80A61ADC.s")
+            case 0x1D05:
+            case 0x1D0C:
+                this->watchTarget = EN_BAL_WATCHTRG_FAIRY;
+                Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimationInfo, TINGLE_ANIM_HAPPY_DANCE_LOOP);
+                Message_StartTextbox(play, 0x1D0D, &this->picto.actor);
+                this->prevTextId = 0x1D0D;
+                break;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bal/func_80A61B5C.s")
+            case 0x1D06:
+            case 0x1D17:
+                this->watchTarget = EN_BAL_WATCHTRG_NONE;
+                Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimationInfo, TINGLE_ANIM_MAGIC_FULL);
+                this->forceEyesShut = true;
+                Message_StartTextbox(play, 0x1D07, &this->picto.actor);
+                this->prevTextId = 0x1D07;
+                break;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bal/func_80A61C9C.s")
+            case 0x1D07:
+                if (CHECK_WEEKEVENTREG(WEEKEVENTREG_TALKED_TO_TINGLE)) {
+                    Message_CloseTextbox(play);
+                    EnBal_SetupGroundIdle(this);
+                } else {
+                    this->forceEyesShut = true;
+                    this->watchTarget = EN_BAL_WATCHTRG_PLAYER;
+                    Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimationInfo, TINGLE_ANIM_TALK_IDLE);
+                    Message_StartTextbox(play, 0x1D08, &this->picto.actor);
+                    this->prevTextId = 0x1D08;
+                    SET_WEEKEVENTREG(WEEKEVENTREG_TALKED_TO_TINGLE);
+                    EnBal_SetRecognizedPlayerForm();
+                }
+                break;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bal/func_80A61CE4.s")
+            case 0x1D08:
+            case 0x1D09:
+            case 0x1D0A:
+                Message_CloseTextbox(play);
+                EnBal_SetupGroundIdle(this);
+                break;
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bal/func_80A61DF8.s")
+void EnBal_SetupOfferGetItem(EnBal* this) {
+    this->actionFunc = EnBal_OfferGetItem;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bal/func_80A61E5C.s")
+void EnBal_OfferGetItem(EnBal* this, PlayState* play) {
+    GetItemId mapItemId;
```

Comment:
```suggestion
    GetItemId mapGetItemId;
```
`ItemId` and `GetItemId` are two different enums so it's important to distinguish between them